WordPress/Classes/ViewRelated/Post/Scheduling/SchedulingCalendarViewController.swift

```diff
@@ -0,0 +1,199 @@
+
+            // If we have an existing time value, we want to set it to the calendar's selected date (which starts at midnight)
+            if let existingDate = self?.coordinator?.date {
+                let components = Calendar.current.dateComponents([.hour, .minute], from: existingDate)
+                newDate = Calendar.current.date(bySettingHour: components.hour ?? 0, minute: components.minute ?? 0, second: components.second ?? 0, of: newDate) ?? newDate
+            }
+            self?.coordinator?.setDate(newDate)
+            self?.chosenValueRow.detailLabel?.text = date.longString()
+        }
+
+        self.calendarMonthView = calendarMonthView
+
+        return calendarMonthView
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        chosenValueRow.titleLabel?.text = NSLocalizedString("Choose a date", comment: "Label for Publish date picker")
+
+        let nextButton = UIBarButtonItem(title: NSLocalizedString("Next", comment: "Next screen button title"), style: .plain, target: self, action: #selector(nextButtonPressed))
+        navigationItem.setRightBarButton(nextButton, animated: false)
+
+        calendarMonthView.setContentCompressionResistancePriority(.defaultHigh, for: .vertical)
+        calendarMonthView.setContentHuggingPriority(.defaultHigh, for: .horizontal)
+        calendarMonthView.setContentHuggingPriority(.defaultHigh, for: .vertical)
+    }
+
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        calculatePreferredSize()
+    }
+
+    private func calculatePreferredSize() {
+        let targetSize = CGSize(width: view.bounds.width,
+          height: UIView.layoutFittingCompressedSize.height)
+        preferredContentSize = view.systemLayoutSizeFitting(targetSize)
+    }
+
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        (segue.destination as? DateCoordinatorHandler)?.coordinator = coordinator
+    }
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+
+        let closeButton = UIBarButtonItem(image: Gridicon.iconOfType(.cross), style: .plain, target: self, action: #selector(closeButtonPressed))
+
+        let publishButton = UIBarButtonItem(title: NSLocalizedString("Publish immediately", comment: "Immediately publish button title"), style: .plain, target: self, action: #selector(publishImmediately))
+        navigationItem.setLeftBarButton(publishButton, animated: false)
+
+        if traitCollection.verticalSizeClass == .compact {
+            navigationItem.leftBarButtonItems = [closeButton, publishButton]
+        } else {
+            navigationItem.leftBarButtonItems = [publishButton]
+        }
+    }
+
+    @objc func closeButtonPressed() {
+        dismiss(animated: true, completion: nil)
+    }
+
+    @objc func publishImmediately() {
+        coordinator?.updated(nil)
```

Comment:
So "Coordinator" may be a bad name for this. It's really passing the "internal" `date` value between controllers and then holding a reference to the `updated` callback, which actually sets the date for the settings screen. So `date` needs to change locally throughout the editing process but only notify `PublishSettingsViewController` when complete.

I was thinking these picker screens could be constructed as a sequence with a generic value type that would be output at the end of the process. I ended up just keeping it specific to dates for now to be genericized later if the half screen modal view ends up being used elsewhere.

Maybe Flux is a better fit here? I've been too buried in Rx and need to reset my thinking a little for it.

Reply:
@bjtitus I played a bit with this. A part from the name, you can keep this to handling the selected date but it should be a reference and not a value type, in order to avoid the copy. If you don't like it I think you can solve this using a delegate to inject, a closure to inject or even a protocol that every vc in the hierarchy can use (I made an example [here](https://gist.github.com/danielebogo/c25e7c1c585b9622406a263f0f68074b)).
Flux also can be a solution but I would use it to handle more things than the simple date.
Btw for me it's fine to use this as simple reference 😊